### PR TITLE
fix: filter out `optimism-goerli-staging` from chains

### LINKED ISSUE #58: Fix: remove optimism-goerli-staging from SDK chains
## Summary

If you deploy a table using Optimsim Goerli, it will default to the `optimism-goerli-staging` contract instead of `optimism-goerli`. This is a bug since the staging contract should not be used by any developers externally. See discussion [here](https://linear.app/tableland/issue/STU-176/web-optimism-goerli-staging-is-used-instead-of-optimism-goerli) for more details.

## Details

See this [line](https://github.com/tablelandnetwork/tableland-js/blob/19582eb4fea10388d2659dacdf5bd151cf73c2fe/packages/sdk/src/helpers/chains.ts#L44)—here (or in the preceding code) we must filter out the "staging" keyword. The CLI does this with the following, so something similar must be put into place:

```
export const getChains = function (): typeof helpers.supportedChains {
  return Object.fromEntries(
    Object.entries(helpers.supportedChains).filter(
      ([name]) => !name.includes("staging")
    )
  ) as Record<helpers.ChainName, helpers.ChainInfo>;
};
```

<sub>From [SyncLinear.com](https://synclinear.com) | [JST-35](https://linear.app/tableland/issue/JST-35/fix-remove-optimism-goerli-staging-from-chains)</sub>

### PR DESCRIPTION
**Pull request by dtbuchholz**
## Summary

If you try to create a table on OP Goerli, it actually uses the `optimism-goerli-staging` network instead of `optimism-goerli`. This filters out the "staging" network and adds tests accordingly. Also, I squeezed in a dep bump for `@tableland/sdk`. Closes #58.

## Details

There's a simple one liner added to `helpers/chains.ts`:
```ts
const filtered = mapped.filter(
  ([chainName]) => chainName !== "optimism-goerli-staging"
);
```

This is then used in the `supportedChains` object, which gets exported and is used in clients like the CLI. Note that the CLI had to implement similar logic to what's described above. The change in the SDK will allow us to remove that check in the CLI, but it's not required at the moment; everything should still work as expected in the CLI.

Wrt to the `@tableland/sdk@^4.5.3-dev.0` dep bump, I also deleted `package-lock.json` and reinstalled the overall deps in root. It seems that lerna was having issues with symlinks, and deleting the lock file & updating the packages seemed to fix it. I think lerna might require that each package is using the latest version—e.g., in `packages/cli`, if I don't use `4.5.3-dev.0` but use any earlier SDK version, I cannot set up a symlink successfully.

## How it was tested

Added a test in `chains.test.ts` to make sure that `getChainId("optimism-goerli-staging")` will throw an error for `cannot use unsupported chain`.

I also set up a symlink to the Studio, and I successfully created a table on the _correct_ OP Goerli network.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by dtbuchholz** [comment]
@joewagner i made a small adjustment to the SDK to fix a bug that was uncovered during ETHOnline while someone was using the Studio. it removes `optimism-goerli-staging` from the available chains, and i included some dep chores to resolve lerna symlink issues. lmk if the deps aspect isn't ideal, and i can remove em. cc @asutula

File: packages/cli/src/commands/write.ts [hunk 1]
```diff
@@ -111,9 +111,8 @@ export const handler = async (argv: Arguments<Options>): Promise<void> => {
             norm.tables[0]
           );
 
-          const { tableId } = await globalThis.sqlparser.validateTableName(
-            name
-          );
+          const { tableId } =
+            await globalThis.sqlparser.validateTableName(name);
```
Comment (joewagner) [inline packages/cli/src/commands/write.ts:115]: Looks like the prettier settings are changed.  Is this something to do with the deleted package lock?

File: packages/cli/package.json [hunk 2]
```diff
@@ -54,7 +54,7 @@
   },
   "dependencies": {
     "@ensdomains/ensjs": "3.0.0-alpha.64",
-    "@tableland/sdk": "^4.4.2",
+    "@tableland/sdk": "^4.5.3-dev.0",
```
Comment (joewagner) [inline packages/cli/package.json:57]: Let's bump the sdk to the next appropriate version.

**Comment by joewagner** [comment]
@dtbuchholz this looks good. We might as well put the sdk at an official release number.  Looks like something got messed up during the last publish and npm has the sdk's `latest` version set at `4.5.3-dev.0`, which should be the `next` version.
I can guess how that happened (probably my mistake) but let's fix it with this pr.  I think version `4.5.3` makes sense here since it's considered a bug.

**Review by asutula — APPROVED** [review]

File: packages/cli/src/commands/write.ts (hunk 1 above)
Comment (dtbuchholz) [reply packages/cli/src/commands/write.ts]: yeah, looks like it. i dropped that commit, and the formatting went back to normal. i'll just ignore any of the lerna-related things i included here...it _might_ not be the root cause, anyways, after some additional testing...

**Comment by dtbuchholz** [comment]
> I think version 4.5.3 makes sense here since it's considered a bug.

@joewagner i bumped the versions in the cli & local packages to 4.5.3. but since it doesn't exist, the tests etc obviously fail here. is that fine, or should i move the dep bump to a separate PR?

**Comment by joewagner** [comment]
> @joewagner i bumped the SDK versions in the cli & local packages to 4.5.3. but since it doesn't exist, the tests etc obviously fail here. is that fine, or should i move the dep bump to a separate PR?

> edit: i'll also have to bump the `package/sdk`'s `package.json` version to `4.5.3` too, right? or does the publish GH action somehow handle that? (this change is not included yet, but i'll wait for your input above before i do anything there.)

You should be able to set the version to `4.5.3`, then update the dependencies in the other packages and as long as the build and test works this PR is good to go.  Once merged someone(I can do it, or we can go through it together in voice), can run the lerna publish flow described here: https://www.notion.so/textile/Lerna-Publish-Playbook-55fab12115fd4506926ee1a0556ad63a

**Comment by joewagner** [comment]
@dtbuchholz I accidentally edited your comment instead of replying...
I tried to put it back how it was, but I'm not totally sure.
I really don't understand why github lets you edit someone else's comment

**Comment by dtbuchholz** [comment]
> @dtbuchholz I accidentally edited your comment instead of replying... I tried to put it back how it was, but I'm not totally sure. I really don't understand why github lets you edit someone else's comment

@joewagner no worries...that's such a dumb feature lol

> Once merged someone(I can do it, or we can go through it together in voice), can run the lerna publish flow described here: https://www.notion.so/textile/Lerna-Publish-Playbook-55fab12115fd4506926ee1a0556ad63a

i'm down to take a stab! i'll ping you on discord since i know timezones are a bit tough rn

**Review by joewagner — APPROVED** [review]

**Pull request merged**